featurebyte/query_graph/sql/ast/input.py

```diff
@@ -99,6 +100,35 @@ def _maybe_cast(identifier_name: str) -> Expression:
 
         return select_expr
 
+    def _select_from_dbtable(self, select_expr: Select, dbtable: Expression) -> Select:
+        on_demand_entity_filters = self.context.on_demand_entity_filters
+        if (
+            on_demand_entity_filters is not None
+            and self.context.parameters["id"] in on_demand_entity_filters.mapping
+        ):
+            original_cols = [
+                quoted_identifier(col_info["name"])
+                for col_info in self.context.parameters["columns"]
+            ]
+            entity_filter = on_demand_entity_filters.mapping[self.context.parameters["id"]]
+            # Need to deduplicate entity table if entity_filter only uses a subset of the entity
+            # columns
```

Comment:
If i understood it correctly, does this mean the table only contains subset but not full of the composite entity columns?

Reply:
yes, the table only contains a subset of the columns.